app/Http/Controllers/PreferenceController.php

```diff
@@ -24,10 +25,26 @@ public function update(Request $request)
             collect($request->only(Preferences::getValidKeys()))->filter()
         );
 
-        if (auth()->user() && $request->filled('track')) {
-            auth()->user()->track_id = $request->input('track');
+        $user = auth()->user();
 
-            auth()->user()->save();
+        if ($user) {
+            if ($request->filled('track')) {
+                $user->track_id = $request->input('track');
+            }
+
+            $request->validate([
+                'is_subscriber' => 'nullable|boolean',
+            ]);
+
+            $user->is_subscriber = $request->has('digest-subscriber');
+
+            if ($user->is_subscriber) {
+                $user->unsubscribe_token = Str::random(60);
+            } else {
+                $user->unsubscribe_token = null;
+            }
```

Comment:
Since the `unsubscribe_token` is set either way you can replace the conditional with a ternary.

```suggestion
            $user->unsubscribe_token = $user->is_subscriber ? Str::random(60) : null;
```